File: apps/api/app/api/routes/credit_file.py
from fastapi import APIRouter

from app.schemas.credit_file import CaseInboxResponse, CreditFileResponse, EvidenceMapResponse
from app.services.credit_file_service import get_case_inbox, get_credit_file, get_evidence_map
from app.services.transaction_summary_service import TransactionSummary, get_transaction_summary

router = APIRouter(tags=["credit-file"])


@router.get("/credit-file/{msme_id}", response_model=CreditFileResponse)
def credit_file(msme_id: str) -> CreditFileResponse:
    return get_credit_file(msme_id)


@router.get("/credit-file/{msme_id}/evidence-map", response_model=EvidenceMapResponse)
def evidence_map(msme_id: str) -> EvidenceMapResponse:
    return get_evidence_map(msme_id)


@router.get("/credit-file/{msme_id}/transaction-summary", response_model=TransactionSummary)
def transaction_summary(msme_id: str) -> TransactionSummary:
    return get_transaction_summary(msme_id)


@router.get("/case-inbox", response_model=CaseInboxResponse)
def case_inbox() -> CaseInboxResponse:
    return get_case_inbox()

File: apps/web/src/lib/api/credit-file.ts
import { apiFetch } from "./client";
import { API_BASE_URL } from "@/lib/constants";
import {
  caseInboxResponseSchema,
  copilotChatResponseSchema,
  creditFileSchema,
  evidenceMapResponseSchema,
  evidenceRecordSchema,
  type CaseInboxResponse,
  type CopilotChatResponse,
  type CreditFile,
  type EvidenceMapResponse,
  type EvidenceRecord
} from "@/lib/schemas/credit-file";
import { z } from "zod";

export function getCaseInbox(): Promise<CaseInboxResponse> {
  return apiFetch("/case-inbox", caseInboxResponseSchema);
}

export function getCreditFile(msmeId: string): Promise<CreditFile> {
  return apiFetch(`/credit-file/${msmeId}`, creditFileSchema);
}

export function getEvidenceMap(msmeId: string): Promise<EvidenceMapResponse> {
  return apiFetch(`/credit-file/${msmeId}/evidence-map`, evidenceMapResponseSchema);
}

export function getEvidenceRecords(msmeId: string): Promise<EvidenceRecord[]> {
  return apiFetch(`/credit-file/${msmeId}/evidence`, z.array(evidenceRecordSchema));
}

export function getEvidenceRecord(msmeId: string, evidenceId: string): Promise<EvidenceRecord> {
  return apiFetch(`/credit-file/${msmeId}/evidence/${evidenceId}`, evidenceRecordSchema);
}

export function updateEvidenceStatus(msmeId: string, evidenceId: string, status: string): Promise<EvidenceRecord> {
  return apiFetch(`/credit-file/${msmeId}/evidence/${evidenceId}/status`, evidenceRecordSchema, {
    method: "PATCH",
    body: { status }
  });
}

export function uploadEvidence(msmeId: string, file: File, sourceType = "uploaded_document", status = "partial") {
  const formData = new FormData();
  formData.set("file", file);
  formData.set("source_type", sourceType);
  formData.set("status", status);
  return apiFetch(`/credit-file/${msmeId}/evidence/upload`, z.object({ record: evidenceRecordSchema, audit_event_id: z.string() }), {
    method: "POST",
    formData
  });
}

export function evidenceFileUrl(msmeId: string, evidenceId: string) {
  return new URL(`/credit-file/${msmeId}/evidence/${evidenceId}/file`, API_BASE_URL).toString();
}

export function sendCopilotChat(msmeId: string, message: string, mode?: string): Promise<CopilotChatResponse> {
  return apiFetch(`/copilot/${msmeId}/chat`, copilotChatResponseSchema, {
    method: "POST",
    body: { message, mode, include_trace: true }
  });
}
